ringpop.go

```diff
@@ -164,25 +171,29 @@ func (rp *Ringpop) identity() (string, error) {
 // TChannel object on the Ringpop instance.
 func (rp *Ringpop) channelIdentityResolver() (string, error) {
 	hostport := rp.channel.PeerInfo().HostPort
+	// Check that TChannel is listening. By default, TChannel listens on an
+	// ephemeral host/port. The real port is then assigned by the OS when
+	// ListenAndServe is called. If the hostport is 0.0.0.0:0, it means
+	// TChannel is not yet listening and the hostport cannot be resolved.
+	if hostport == "0.0.0.0:0" {
+		return "", fmt.Errorf("unable to resolve valid listen address (TChannel hostport is %s)", hostport)
+	}
 	return hostport, nil
 }
 
 // Destroy Ringpop
 func (rp *Ringpop) Destroy() {
-	rp.node.Destroy()
 
-	rp.state.Lock()
-	rp.state.destroyed = true
-	rp.state.Unlock()
-}
+	if rp.node != nil {
```

Comment:
This is what I had initially, but I changed it to the current code. That is, if `rp.node` is not `nil`, call `Destroy()`.

> Otherwise, you'd then be dependent upon whether rp.node.Destroy() were idempotent too.

`rp.node.Destroy()` *is* idempotent, as it should be (IMHO). The test passes. This seems like the correct behaviour to me.